react/features/connection-indicator/components/AbstractConnectionIndicator.js

```diff
@@ -0,0 +1,179 @@
+// @flow
+
+import { Component } from 'react';
+
+import statsEmitter from '../statsEmitter';
+
+declare var interfaceConfig: Object;
+
+/**
+ * The connection quality percentage that must be reached to be considered of
+ * good quality and can result in the connection indicator being hidden.
+ *
+ * @type {number}
+ */
+export const INDICATOR_DISPLAY_THRESHOLD = 30;
+
+/**
+ * The type of the React {@code Component} props of {@link ConnectionIndicator}.
+ */
+export type Props = {
+
+    /**
+     * The ID of the participant associated with the displayed connection indication and
+     * stats.
+     */
+    participantId: string
+};
+
+/**
+ * The type of the React {@code Component} state of {@link ConnectionIndicator}.
+ */
+export type State = {
+
+    /**
+     * The timeout for automatically hiding the indicator.
+     */
+    autoHideTimeout: ?TimeoutID,
```

Comment:
I don't think autoHideTimeout should be on state unless it really does need to trigger a re-render. I feel it might be better as a direct instance variable, a this._autoHideTimeout. So why did I put it on there?

Reply:
You're right. Not sure why it was there, I didn't sanity check this code part :/